deployer/orb.yaml

```diff
@@ -1,6 +1,85 @@
 version: 2.1
 description: "Tools for running deployment commands"
 
+commands:
+  validate-tag-for-pip:
+    description: >
+      Uses pips own Version parser to ensure semver semantics of the tag match
+      exactly with those used by pip.
+    parameters:
+      verify_version:
+        default: CIRCLE_TAG
+        description: >
+          The string or string-name of env var which holds the version string
+          to parse with pip Version
+        type: string
+      should_upgrade_pip:
```

Comment:
In what case would we not want to do this?

Perhaps it would be more useful to accept a version specifier here and do eg.
```yaml
parameters:
  pip_version:
    default: ''
    type: string
    description: >
      Pip version specifier. Defaults to the latest version, set 
      to '==1.2.3' or '>=42.0, <50' if you have more specific
      requirements!
steps:
- run: python3 -m pip install --upgrade pip<<parameters.pip_version>>
```